src/MICore/Transports/PipeTransport.cs

```diff
@@ -137,6 +137,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             proc.StartInfo.Arguments = pipeOptions.PipeArguments;
             proc.StartInfo.WorkingDirectory = pipeOptions.PipeCwd;
 
+            foreach (EnvironmentEntry entry in pipeOptions.PipeEnvironment)
+            {
+                proc.StartInfo.SetEnvironmentVariable(entry.Name, entry.Value);
```

Comment:
If I remember correctly (and maybe I don't) this is missing from Mono, and you want to use StartInfo.Environment.Add